src/server/bg_services/log_replication_scanner.js

```diff
@@ -111,70 +114,74 @@ class LogReplicationScanner {
      * @param {*} dst_bucket destination bucket
      * @param {nb.ReplicationLogCandidates} candidates
      * @param {boolean} sync_versions should we sync object versions
+     * @param {boolean} sync_deletions should we sync deletions
      * @param {string} rule_id
      * @param {string} replication_id
      */
-    async process_candidates(src_bucket, dst_bucket, candidates, sync_versions, rule_id, replication_id) {
+    async process_candidates(src_bucket, dst_bucket, candidates, sync_versions, sync_deletions, rule_id, replication_id) {
         let copy_keys;
         let delete_keys;
         if (sync_versions) {
-            ({ copy_keys, delete_keys } = await this.process_candidates_sync_version(src_bucket, dst_bucket, candidates));
+            copy_keys = await this.process_candidates_sync_version(src_bucket, dst_bucket, candidates, false);
+            // for sync_versions enabled, deletions cannot be performed until the copying process is completed
+            await this.copy_objects(src_bucket.name, dst_bucket.name, copy_keys, rule_id, replication_id);
+
+            if (sync_deletions) { // If sync_deletions is enabled, then only the deletion keys for versioned objects are captured
+                const diff_keys = await this.process_candidates_sync_version(src_bucket, dst_bucket, candidates, sync_deletions);
+                delete_keys = Object.keys(diff_keys); // fetching keys array from diff_keys
+                await this.delete_objects(dst_bucket.name, delete_keys);
+            }
         } else {
             ({ copy_keys, delete_keys } = await this.process_candidates_not_sync_version(src_bucket, dst_bucket, candidates));
```

Comment:
@liranmauda Do we need to put comment like:

```suggestion
            // here even if sync_deletions is disabled, we are processing delete_keys for not_sync_verison
            ({ copy_keys, delete_keys } = await this.process_candidates_not_sync_version(src_bucket, dst_bucket, candidates));
```

Reply:
@Neon-White wdyt?

Reply:
I am not sure, but may be it is the required to call both `copy_objects` and `delete_objects` in parallel for not_sync_version.

Reply:
Putting the above comment for now. Thanks